app/models/results/response_tree_builder.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Results
+  # Builds a response tree from a form object
+  class ResponseTreeBuilder
+    def initialize(form)
+      @form = form
+    end
+
+    def build
+      @rn = AnswerGroup.new
```

Comment:
Better to not use an abbreviated var name like this. `root` works. Also I don't think it needs to be an ivar.

Reply:
done